src/server.cpp

```diff
@@ -1352,6 +1358,10 @@ void CServer::ProcessData ( const CVector<CVector<int16_t> >& vecvecsData,
             // get a reference to the audio data and gain of the current client
             const CVector<int16_t>& vecsData = vecvecsData[j];
             const double            dGain    = vecdGains[j];
+            const double			dPan = vecdPannings[j];
+            const double			dPanCoefL = sqrt(1-dPan); // faster: (1-dPan)
```

Comment:
If the dPan is 0.5, then we have sqrt(0.5) = 0.7. So we would get an attenuation. Is that wanted?

Reply:
Yes, this sounds better. If we use linear calculation and in case pan=0.5, it sounds (seems) weaker when sound is in the center than on the edges. This is a bit unnatrural. The sqrt balances it a bit, so on pan=0.5 both channels get 0.7 of the power and that seems more real. Even better would be to use a calulation like:
gainL =  sin((dPan + 0.5) * $M_PI*2) * dGain
gainR = cos((dPan + 0.5) * $M_PI*2) * dGain
(then the maximum power is a bit before the hard left/right)
but i think it is not so important here and we can save some computation power.

Reply:
Don't know if this is of any help:

If pan is centre (0.5) and gain is unattenuated (1), then the value to multiply by should be 1 for both channels - and the optimiser should skip doing anything.

Unattentuated hard left and right should sum to -6dB for mono - i.e. multiply by 0.5 on one side, 0 on the other.  (Stereo pan should be mixing further to mono and ending up about -4.5dB on either channel, if I remember theory correctly..., that's the 0.7.)

Reply:
Ok, but my concern is that after adding the pan feature the signal amplitude is lower than it was before that change. Usually the people will have the pan in the middle. Then they may complain that the level has changed by the new version of Jamulus.

Reply:
I understand it now that it influences especially situation when there are two different signals on different channels (like vocals in one and guitar in other). I think the solution would be to have an enable/disable button close to channel pan slider so user can switch it out easily (by channel). If panning is out, then both channels would be multiplied with 1, ie no change. Otherwise if to use panning, the situation is different from previous experience anyway, so I would not worry as long the panning effect sound fine.

Reply:
I will get to the changes and hope to submit new pull request next week. Thanks!

Reply:
I would not add another button. The faders should have as little GUI controls as possible.
As an initial implementation we accept the level drop (actually it is not too bad, maybe it is not really noticeable).